llama_index/agent/react/base.py

```diff
@@ -336,27 +364,47 @@ def stream_chat(
             chat_stream = self._llm.stream_chat(input_chat)
 
             # iterate over stream, break out if is final answer after the "Answer: "
-            is_done = False
             full_response = ChatResponse(
                 message=ChatMessage(content=None, role="assistant")
             )
-            for r in chat_stream:
-                if "Answer: " in (r.message.content or ""):
-                    is_done = True
-                    break
-                full_response = r
-            if is_done:
-                break
+            for latest_chunk in chat_stream:
+                latest_content = latest_chunk.message.content
+                if latest_content:
+                    if not latest_content.startswith(
+                        "Thought"
+                    ):  # doesn't follow thought-action format
+                        is_done = True
+                        full_response = latest_chunk
+                        break
+                    else:
+                        if "Answer: " in latest_content:
+                            is_done = True
+                            full_response = latest_chunk
+                            break
+                    full_response = latest_chunk
```

Comment:
very nit: can we pull this out to separate function to dedup.

Reply:
good idea! done.